src/main/java/com/moneytransfer/aspect/IdempotentTransferAspect.java

```diff
@@ -0,0 +1,162 @@
+     */
+    private Request getOrSubmitRequest(final UUID requestId, final NewTransferDto newTransferDto) throws MoneyTransferException {
+        Optional <Request> optionalRequest = Optional.ofNullable(requestService.getRequest(requestId));
+        return optionalRequest.isPresent() ? optionalRequest.get() : requestService.submitRequest(requestId,newTransferDto);
+    }
+
+    /**
+     * Processes the transfer and resolves the Request.
+     *
+     * @param request
+     * @param proceedingJoinPoint
+     * @return a new Transaction
+     * @throws Throwable
+     */
+    private Transaction processTransfer(final Request request, final ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        Transaction transaction = null;
+        try {
+            transaction = (Transaction) proceedingJoinPoint.proceed();
+            return transaction;
+        } catch (MoneyTransferException e) {
+            transaction = persistFailedTransfer(request, e);
+            throw e;
+        } finally {
+            requestService.resolveRequest(request.getRequestId(), transaction);
+        }
+    }
+
+    /**
+     * Retrieves the Transaction of a RESOLVED Request.
+     *
+     * @param request
+     * @return the persisted Transaction
+     * @throws Throwable
+     */
+    private Transaction retrieveTransaction(final Request request) throws Throwable {
+        Transaction transaction = request.getTransaction();
+        return switch (transaction.getTransactionStatus()) {
+            case SUCCESSFUL_TRANSFER -> transaction;
+            case FAILED_TRANSFER ->
+                    throw new RequestConflictException(transaction.getMessage(), transaction.getHttpStatus());
+        };
+    }
+
+    /**
+     * Persists the failed transfer.
+     *
+     * @param request
+     * @param e
+     * @return a new failed Transaction
+     * @throws ResourceNotFoundException
+     */
+    private Transaction persistFailedTransfer(final Request request, final MoneyTransferException e) throws ResourceNotFoundException {
+        var targetAccount = getAccountService.getAccountByIdOrReturnDefault(request.getTargetAccountId());
+        var sourceAccount = getAccountService.getAccountByIdOrReturnDefault(request.getSourceAccountId());
+        var httpStatus = e.getHttpStatus();
+        var errorMessage = e.getMessage();
+        var currency = sourceAccount.getCurrency();
+        return transactionRepository.save(new Transaction(UUID.randomUUID(), TransactionStatus.FAILED_TRANSFER, sourceAccount, targetAccount, request.getAmount(), errorMessage, currency, httpStatus));
+    }
+
+
+    /**
+     * Validates payload idempotency.
+     *
+     * @param request
+     * @param newTransferDto
+     * @throws RequestConflictException
+     */
+    private void validateIdempotent(final Request request, final NewTransferDto newTransferDto) throws RequestConflictException {
+        if (newTransferDto.hashCode() != request.toNewTransferDto().hashCode()) {
```

Comment:
Comparing hashCodes of different objects.... Better create a method in the aspect that does the comparison...